src/xrScriptEngine/script_engine.cpp

```diff
@@ -754,8 +753,20 @@ void CScriptEngine::lua_error(lua_State* L)
 
 int CScriptEngine::lua_pcall_failed(lua_State* L)
 {
-    print_output(L, "", LUA_ERRRUN);
+#if (!defined(DEBUG) && !XRAY_EXCEPTIONS)
+    print_output(L, "", 0); // Force game to not break in raii_guard() for this type of errors
+#else
+    print_output(L, "", LUA_ERRRUN); // Default behavior
+#endif
```

Comment:
OK, here's what I got so far:

- `!XRAY_EXCEPTIONS` is effectively equal to `!defined(DEBUG)`.
- luabind uses `pcall()` to execute lua functions. Each time `pcall()` is called, pcall_callback is passed to luajit by luabind. When lua function fails, luajit calls pcall_callback.
- When `pcall()` fails (e.g. lua function fails), luabind calls `luabind::detail::call_error()`, which in it's turn calls error callback (`CScriptEngine::lua_error()`) followed by `assert()` and `std::terminate()`. So, in release configuration the crash occurs in `CScriptEngine::lua_error()`, and in debug configuration it happens in `CScriptEngine::lua_pcall_failed()`. Even when neither of these two callbacks crashes the game, luabind crashes the game by calling `assert()` and `std::terminate()`.
- luabind does not use error callback (see `luabind::get_error_callback()`) in release, which is the exact reason why we can not handle all errors in `CScriptEngine::lua_error()`.

Hope that helps to understand this mess.